### PR TITLE
Add a test for saving editors to a manuscript

### LINKED ISSUE #2176: Add API test for saving teams to manuscript
## Task
We've had problems with submissions becoming corrupted because of changes not being properly persisted to the database, we should write a test to ensure this doesn't happen again, the perfect place for this test is in the new API tests

This is probably a good candidate for knowledge sharing and/or pairing

### PR DESCRIPTION
**Pull request by petereast**
#### Background

What does this PR do?
closes #2176 

#### Any relevant tickets

Please use the phrase 'Closes' if it will fully complete the ticket.

#### How has this been tested?
**Reviewers** ensure that these test requirements are also reviewed.

Please indicate the need for this change to be tested. If not please justify, if so then at what levels. 

- [ ] Unit / Integration tests
- [ ] Browser tests
- [ ] End2end tests

#### UI Changes

- [ ] Has been reviewed against Designs
- [ ] Necessary updates to styleguide

### CODE REVIEW

**Comment by petereast** [comment]
**To reviewer**: The commit 3c4b71f adds quite a bit of formatting noise to the code, all the main changes are in b01f58e which you can review in isolation

File: tools/api-tests/src/api-test/manuscript.test.ts [hunk 1]
```diff
@@ -2,79 +2,200 @@ import { test, TestContext } from "ava-ts";
...
+  // and aren't used anywhere, so we strip them out to check that the stuff we depend on
+  // is right
+  t.deepEqual(
+    stripTypeNameFromJson(savedManuscript.updateSubmission),
+    stripTypeNameFromJson(expectedUpdatedManuscript),
+    "the manuscript that is saved is the same as the original manuscript",
+  );
+});
+
+test("The user can save a the editors on a submission and they're persisted in the database", async (t: TestContext) => {
```
Comment (erezmus) [inline tools/api-tests/src/api-test/manuscript.test.ts]: should be 'can save the editors'

**Review by erezmus — CHANGES_REQUESTED** [review]

File: tools/api-tests/src/api-test/manuscript.test.ts (hunk 1 above)
Comment (petereast) [reply tools/api-tests/src/api-test/manuscript.test.ts]: Much more consise

**Review by erezmus — APPROVED** [review]

File: tools/api-tests/src/api-test/manuscript.test.ts [hunk 2]
```diff
@@ -2,79 +2,197 @@ import { test, TestContext } from "ava-ts";
 import dashboard from "../xpub-api/dashboard";
 import submission from "../xpub-api/submission";
 import { defaultConfig } from "../xpub-api";
-import { Manuscript, ManuscriptInput} from "../generated/graphql";
-import { stripTypeNameFromJson  } from '../utils';
+import { Manuscript, ManuscriptInput } from "../generated/graphql";
+import { stripTypeNameFromJson } from "../utils";
```
Comment (Cooryd) [inline tools/api-tests/src/api-test/manuscript.test.ts:6]: Not an issue, more a consistency thing I've just spotted, we're using single quotes elsewhere in the app so should probably stick to those?

File: tools/api-tests/src/api-test/manuscript.test.ts [hunk 3]
```diff
@@ -2,79 +2,197 @@ import { test, TestContext } from "ava-ts";
...
+  // NOTE: The server adds __typename properties for use internally, we don't need to assert on them so I've
+  // removed them
+  t.deepEqual(
+    stripTypeNameFromJson(expectedUpdatedManuscript),
+    stripTypeNameFromJson(savedManuscript.updateSubmission),
+    "the saved manuscript contains the correct teams data",
+  );
+
+  t.deepEqual(
+    // Huh?: why do we get suggestions back with this one and not the other request?
```
Comment (Cooryd) [inline tools/api-tests/src/api-test/manuscript.test.ts:193]: Has this been raised as an investigation issue if its not intended functionality?

**Review by Cooryd — APPROVED** [review]
Codes good, raised some questions on future work.

File: tools/api-tests/src/api-test/manuscript.test.ts (hunk 2 above)
Comment (petereast) [reply tools/api-tests/src/api-test/manuscript.test.ts:6]: Ah yes, this subfolder is using it's own prettier config, I'm going to add tickets to integrate it more with the rest of the codebase, such as including it in CI, syncing the prettier, eslint (well, tslint) etc. with the main project

File: tools/api-tests/src/api-test/manuscript.test.ts (hunk 3 above)
Comment (petereast) [reply tools/api-tests/src/api-test/manuscript.test.ts:193]: Not quite yet, will add now :+1:

**Pull request merged**